zk-desktop.el

```diff
@@ -234,7 +234,7 @@ To quickly change this setting, call `zk-desktop-add-toggle'."
         ;; replace titles
         (goto-char (point-min))
         (let* ((zk-alist (zk--alist))
-               (ids (zk--id-list)))
+               (ids (zk--id-list nil zk-alist)))
```

Comment:
but if str is nil, then `zk--alist` function is never called, and not used; the function just goes straight to `zk--parse-file`

Reply:
That was the point of [d6be688](https://github.com/localauthor/zk/pull/61/commits/d6be688ea0280576875a262e2883282d30779fe5): if `zk-alist` is passed to `zk--id-list`, no matter whether `str` is nil or not, the function uses the passed `zk-alist` rather than redoing all the work with `zk--parse-file`.